api/envoy/extensions/http/original_ip_detection/xff/v3/xff.proto

```diff
@@ -17,10 +20,42 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 //
 // [#extension: envoy.http.original_ip_detection.xff]
 message XffConfig {
-  // The number of additional ingress proxy hops from the right side of the
-  // :ref:`config_http_conn_man_headers_x-forwarded-for` HTTP header to trust when
-  // determining the origin client's IP address. The default is zero if this option
-  // is not specified. See the documentation for
-  // :ref:`config_http_conn_man_headers_x-forwarded-for` for more information.
-  uint32 xff_num_trusted_hops = 1;
+  oneof xff_trusted_config {
```

Comment:
It's not okay to move a pre-existing field into a oneof, since that's a breaking API change for golang.

Also, the [xDS API style guide](https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md) says not to use oneof anyway.  Instead, simply add the new field next to the existing field, and document the fact that if both are set, the new field takes precedence.

Reply:
Thanks, Mark. I did miss that API style document.

I'll work on updating this.

Reply:
Updated.